proposed/building-arch-specific-applications.md

```diff
@@ -0,0 +1,835 @@
+
+Maintain framework-dependent architecture-neutral applications as the default for `build`.
+  
+* Produces an application with native dependencies for all supported architectures.
+* Remove the capability to produce EXEs for this type of build.
+* If EXEs are desired for this build type, consider for a later release. It's a hard scenario to get right due to filename conflicts.
+
+The ASP.NET and .NET Core console application templates use this default configuration (meaning, not specifying any additional configuration).
+
+#### Framework-dependent architecture-specific applications
+
+Expose a new opt-in build type, called architecture-specific framework-dependent.
+
+* Produces an application with native dependencies and an EXE for a single specified architecture.
+* This build type can be selected via
+  * MSBuild: `<ArchitectureSpecific>true</ArchitectureSpecific>`
+  * CLI: `--architecture-specific` or `-a`
+* A RID must be specified (or it is an error). Any of the following options is legal to specify a RID:
+  * MSBuild: `<RuntimeIdentifier>RID</RuntimeIdentifier>`
+  * CLI: `--runtime RID`, or `-r RID`
+  * MSBuild: `<RuntimeIdentifier>$(NETCoreSdkRuntimeIdentifier)</RuntimeIdentifier>` (means: use RID of SDK; this already works)
+* With this build type, `--self-contained` is `false` by default, even when a RID is specified. `--self-contained` or `--self-contained true` must be specified to produce a self-contained build.
+
+The WPF and Windows Forms application templates will opt-in to this configuration, and will choose Windows 64-bit as their default architecture. These templates will be updated to include the following properties:
+
+```xml
+<ArchitectureSpecific>true</ArchitectureSpecific>
+<RuntimeIdentifier>win-x64</RuntimeIdentifier>
+```
+
+Note: It is possible that the RID should be specified a different way that better aligns with the Visual Studio configuration manager. This document doesn't take an opinion on that. It is a detail to work out.
+
+Alternative idea for specifying architecture:
+
+* MSBuild: `<ArchitectureSpecific>RID</ArchitectureSpecific>`
+
+This alternative idea requires one instead of two lines, but also replaces the existing `RuntimeIdentifier` property for only architecture-specific apps, which is likely confusing.
+
+#### Self-contained architecture-specific applications
+
+Maintain the existing self-contained behavior as-is for .NET Core 2.2 apps and earlier, but change it for 3.0 apps and later.
+
+The behavior of specifying a runtime (via `-r`) currently results ins self-contained applications by default. That's a confusing experience given that we consider framework-dependent applications to be the default. The proposed beavhior for framework-dependent architecture-specific apps makes self-contained opt-in. It would be better to do that for all build types, making the overall system easier to understand. We cannot reasonably do that for existing projects, but we can do it for 3.0 and later projects.
```

Comment:
```suggestion
The behavior of specifying a runtime (via `-r`) currently results ins self-contained applications by default. That's a confusing experience given that we consider framework-dependent applications to be the default. The proposed behavior for framework-dependent architecture-specific apps makes self-contained opt-in. It would be better to do that for all build types, making the overall system easier to understand. We cannot reasonably do that for existing projects, but we can do it for 3.0 and later projects.
```